tigervnc fails to compile

```
ixos-upgrade-start[2034]: building '/nix/store/hz0m1vx241ad7jmpr6622z0g7wa73yww-nixos-manual.desktop.drv'...
ixos-upgrade-start[2034]: builder for '/nix/store/0l4663xbxbm2fa1sh26rbmmgcdj7gdzn-tigervnc-1.10.1.drv' failed with exit code 2; last 1>
ixos-upgrade-start[2034]:     CC       Xvnc-stubs.o
ixos-upgrade-start[2034]:     CXXLD    Xvnc
ixos-upgrade-start[2034]:   /nix/store/x5m45fcnky99r0k41kmdwmjb7zw5k4z4-binutils-2.31.1/bin/ld: ../../os/.libs/libos.a(inputthread.o): >
ixos-upgrade-start[2034]:   /build/source/unix/xserver/os/inputthread.c:321: undefined reference to `ddxInputThreadInit'
ixos-upgrade-start[2034]:   collect2: error: ld returned 1 exit status
ixos-upgrade-start[2034]:   make[2]: *** [Makefile:846: Xvnc] Error 1
ixos-upgrade-start[2034]:   make[2]: Leaving directory '/build/source/unix/xserver/hw/vnc'
ixos-upgrade-start[2034]:   make[1]: *** [Makefile:624: all-recursive] Error 1
ixos-upgrade-start[2034]:   make[1]: Leaving directory '/build/source/unix/xserver/hw'
ixos-upgrade-start[2034]:   make: *** [Makefile:824: all-recursive] Error 1
```

--- a/pkgs/tools/admin/tigervnc/default.nix
+++ b/pkgs/tools/admin/tigervnc/default.nix
@@ -23,6 +23,8 @@ stdenv.mkDerivation rec {
 
   inherit fontDirectories;
 
+  patches = [ ./u_xorg-server-1.20.7-ddxInputThreadInit.patch ];
+
   postPatch = ''
     sed -i -e '/^\$cmd \.= " -pn";/a$cmd .= " -xkbdir ${xkeyboard_config}/etc/X11/xkb";' unix/vncserver
     fontPath=

--- a/pkgs/tools/admin/tigervnc/u_xorg-server-1.20.7-ddxInputThreadInit.patch
+++ b/pkgs/tools/admin/tigervnc/u_xorg-server-1.20.7-ddxInputThreadInit.patch
@@ -1,0 +1,21 @@
+Origin: https://build.opensuse.org/package/view_file/X11:XOrg/tigervnc/u_xorg-server-1.20.7-ddxInputThreadInit.patch
+diff -u -p -r tigervnc-1.10.0.old/unix/xserver/hw/vnc/xvnc.c tigervnc-1.10.0/unix/xserver/hw/vnc/xvnc.c
+--- tigervnc-1.10.0.old/unix/xserver/hw/vnc/xvnc.c	2020-01-15 11:19:19.486731848 +0000
++++ tigervnc-1.10.0/unix/xserver/hw/vnc/xvnc.c	2020-01-15 11:37:33.275445409 +0000
+@@ -295,6 +295,15 @@ void ddxBeforeReset(void)
+ }
+ #endif
+ 
++#if INPUTTHREAD
++/** This function is called in Xserver/os/inputthread.c when starting
++    the input thread. */
++void
++ddxInputThreadInit(void)
++{
++}
++#endif
++
+ void ddxUseMsg(void)
+ {
+     vncPrintBanner();
+

